tests/framework/e2e/cluster.go

```diff
@@ -402,6 +403,10 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
 		panic(fmt.Sprintf("Unknown cluster version %v", cfg.Version))
 	}
 
+	if cfg.SnapshotCatchUpEntries > 0 && execPath == BinPath {
```

Comment:
Why check exec path?

Reply:
This is to make sure we do not set the flag for 3.4 binary

Reply:
Should we rather disable test that cannot use the flag? Assume that existing code already does this, Just a note to look into that in the future.

Reply:
Whether a test can use it or not depend on which version is the leader, not inherently to the test itself.

Reply:
> This is to make sure we do not set the flag for 3.4 binary

Please add a comment to explain this.

Reply:
done